application/src/main/java/org/thingsboard/server/service/edge/rpc/processor/TelemetryEdgeProcessor.java

```diff
@@ -96,13 +106,49 @@ public List<ListenableFuture<Void>> processTelemetryFromEdge(TenantId tenantId,
             if (entityData.hasPostTelemetryMsg()) {
                 result.add(processPostTelemetry(tenantId, customerId, entityId, entityData.getPostTelemetryMsg(), metaData));
             }
+            if (EntityType.DEVICE.equals(entityId.getEntityType())) {
+                Device device = deviceService.findDeviceById(tenantId, new DeviceId(entityId.getId()));
+                // for edge context sessionId is exact edgeSessionId
+                reportActivity(device, edgeSessionId);
+            }
         }
         if (entityData.hasAttributeDeleteMsg()) {
             result.add(processAttributeDeleteMsg(tenantId, entityId, entityData.getAttributeDeleteMsg(), entityData.getEntityType()));
         }
         return result;
     }
 
+    private void reportActivity(Device device, UUID sessionId) {
+        TransportProtos.SessionInfoProto.Builder builder = TransportProtos.SessionInfoProto.newBuilder()
+                .setSessionIdMSB(sessionId.getMostSignificantBits())
+                .setSessionIdLSB(sessionId.getLeastSignificantBits())
+                .setTenantIdMSB(device.getTenantId().getId().getMostSignificantBits())
+                .setTenantIdLSB(device.getTenantId().getId().getLeastSignificantBits())
+                .setDeviceIdMSB(device.getId().getId().getMostSignificantBits())
+                .setDeviceIdLSB(device.getId().getId().getLeastSignificantBits())
+                .setDeviceName(device.getName())
+                .setDeviceType(device.getType())
+                .setDeviceProfileIdMSB(device.getDeviceProfileId().getId().getMostSignificantBits())
+                .setDeviceProfileIdLSB(device.getDeviceProfileId().getId().getLeastSignificantBits());
+
+        if (device.getCustomerId() != null && !device.getCustomerId().isNullUid()) {
+            builder.setCustomerIdMSB(device.getCustomerId().getId().getMostSignificantBits());
+            builder.setCustomerIdLSB(device.getCustomerId().getId().getLeastSignificantBits());
+        }
+        reportActivity(device.getTenantId(), device.getId(), device.getUuidId(), builder.build());
+    }
+
+    private void reportActivity(TenantId tenantId, DeviceId deviceId, UUID routingKey, TransportProtos.SessionInfoProto sessionInfo) {
+        TransportProtos.SubscriptionInfoProto subscriptionInfoProto = TransportProtos.SubscriptionInfoProto.newBuilder()
+                .setAttributeSubscription(false).setRpcSubscription(false)
+                .setLastActivityTime(System.currentTimeMillis()).build();
+        TransportProtos.TransportToDeviceActorMsg msg = TransportProtos.TransportToDeviceActorMsg.newBuilder().setSessionInfo(sessionInfo)
+                .setSubscriptionInfo(subscriptionInfoProto).build();
+        TopicPartitionInfo tpi = partitionService.resolve(ServiceType.TB_CORE, tenantId, deviceId);
+        tbCoreMsgProducer.send(tpi, new TbProtoQueueMsg<>(routingKey,
```

Comment:
@volodymyr-babak, I am not sure that this is the best approach. If you just want to know the device active/inactive state, you may "create" new message  that will trigger 
```
systemContext.getDeviceStateService().onDeviceActivity(tenantId, deviceId, System.currentTimeMillis());
```
This will be easier and will consume much fewer resources. Because otherwise, you hardcode many values for the "fake" device session (like attribute and rpc subscriptions). However, if you do need this session to register in the device actor - the code changes are ok. Please provide your comments